flake.nix

```diff
@@ -2,63 +2,117 @@
-          fileAttrs = lib.filterAttrs (_: t: t == "regular") (builtins.readDir ./src);
-          fileList = with builtins; filter (lib.hasSuffix "nix") (attrNames fileAttrs);
-          src = map (file: ./src/${file}) fileList;
+          inherit (nixpkgs.lib) nixosSystem;
+          rootConfig = config;
+          this = (import ./lib.nix args);
+          src = map (file: ./src/${file})
+            (this.walkNixFiles ./src);
         in
         {
           imports = with inputs; [
-            lynx.flakeModules.builtins
-            lynx.flakeModules.flake-guard
-            asluni.flakeModules.asluni
             git-hooks-nix.flakeModule
+            lynx.flakeModules.flake-guard
+            ./check-users.nix
           ];
 
           config = {
             systems = [ "x86_64-linux" ];
+            flake = {
+              ##################
+              # intended for
+              # note: `nix repl`
+              lib = lib.fix (
+                lib.extends
+                  (f: p: this)
+                  (f: lib)
+              );
+              ##################
+
+              modules.nixos = with inputs; [
+                disko.nixosModules.disko
+                lynx.nixosModules.flake-guard-host
+                asluni.nixosModules.asluni
+                sops.nixosModules.sops
+                { environment.etc.nixos.source = self; }
+                { environment.etc.nixpkgs.source = nixpkgs; }
+              ] ++ src;
+
+              nixosConfigurations.helium = this.buildBox {
+                specialArgs = { inherit self inputs; };
+                modules = config.flake.modules.nixos;
+              };
+            };
 
             perSystem = { config, pkgs, ... }: {
               pre-commit.check.enable = true;
 
               pre-commit.settings.hooks = {
                 trim-trailing-whitespace.enable = true;
                 nixpkgs-fmt.enable = true;
+                flake-checker.enable = true;
               };
 
               devShells.default = pkgs.mkShell {
                 shellHook = lib.concatStringsSep "\n" [
                   config.pre-commit.installationScript
+                  ''
+                    sops-recrypt() {
+                      sops decrypt $1 | sops encrypt --filename-override ''${2:-$1} /dev/stdin
+                    }
+                  ''
                 ];
                 packages = with pkgs; [
+                  sops
+                  git-extras
+                  git-bug
+                  git
                   pre-commit
                 ];
               };
-            };
 
-            flake = {
-              nixosConfigurations.helium = lib.nixosSystem {
-                specialArgs = { inherit inputs self; };
-                modules = src ++ [
-                  inputs.disko.nixosModules.disko
-                ];
+              checks.helium = pkgs.testers.runNixOSTest {
+                name = "helium";
+                node.pkgsReadOnly = false;
```

Comment:
why do we have this set to false? issues with nixos test?

Reply:
pkgsReadOnly assumes no configurations can be done to nixpkgs. eg, allow unfree cannot be used